apps/test-stress-can/src/main.cpp

```diff
@@ -0,0 +1,82 @@
+#include "CANBus.h"
+#include "CANMsg.h"
+#include "mbed.h"
+
+void implementTraditionalMethod();
+void implementInterruptMethod();
+
+CANBus can(CAN_RX, CAN_TX, HWBRIDGE::ROVERCONFIG::ROVER_CANBUS_FREQUENCY);
+
+// create an event queue
+EventQueue queue;
+
+// This variable isn't made locally because the implementInterruptMethod function needs access to it to print the number
+// of messages received
+uint16_t ISRReceived = 0;
+
+// create a Timer to limit the length that the process can take
+Timer timer;
+
+void CANReceiveEventHandler(CANMsg &msg) {
+  // Print data received from the CAN msg
+  printf("  Data    =");
+  for (int i = 0; i < msg.len; i++) printf(" 0x%.2X", msg.data[i]);
+  printf("\r\n");
+
+  ISRReceived++;
+}
+
+void CANMsgIRQHandler() {
+  CANMsg msg;
+  if (can.read(msg)) {
+    queue.call(&CANReceiveEventHandler, msg);
+
+  } else {
+    printf("THIS SHOULD NEVER HAPPEN \r\n");
+    MBED_ASSERT(false);
+  }
+}
+
+int main() {
```

Comment:
I don't know where to write this but i felt like its part of the scope of this pr. 

we should investigate tderror and rderror to see exactly what it does and probably incorporate it into whatever can library we end up writing out of all this

Reply:
its important to keep the tasks manageable in size and not have em grow too much. this pr is just to prove that we drop less packets by using interrupts and a semaphore over the old method we had which received _and_ processed msgs in the same 'breath'

Reply:
Fair but I mean all I'm really saying is maybe add an if check on tderror/terror terror. I guess its not part of this test or but the actual can library that comes out of this